<!DOCTYPE html>
<html>
<head>
<!DOCTYPE html>
<html lang="en-gb">



<link rel="stylesheet" href="/css/fonts.css">
<link rel="stylesheet" href="/css/main-site.css">
<link rel="stylesheet" href="/css/fa5-all.css">
<style type="text/css">
    /* Google Fonts */
 @import url('https://fonts.googleapis.com/css2?family=Libre+Baskerville:ital,wght@0,400;0,700;1,400&display=swap'); 

body {
	background-color: white;
	color: #404040;
}

/* Links */

a {
  color: #4254A7;
}

a:hover {
  color: #4254A7;
}

a:focus {
  outline-color: #4254A7;
}

.column25-left .sectionTitle a:hover:not(.current),
.column16-left .sectionTitle a:hover:not(.current) { /* Tutorial left sidebar */
  color: #404040;
}

.icon-attribution, 
.icon-attribution a {
  color: #404040;
  opacity: 75%;
}


/* Landing page content bands */

#homeContent .band.first {
  background-color: white;
}

#homeContent .band.second {
  background-color: #fcfcfc;
}

#homeContent .band.third {
  background-color: white;
}

/* Top navigation bar menu */

#rStudioHeader {
  background-color: white;
  color: black;
}

#rStudioHeader .productName {
  color: #4254A7;
}

#rStudioHeader .productName:hover,
#rStudioHeader .productName:focus {
  color: 00000075;
}

#rStudioHeader #menu .menuItem.a {
  background-color: #75aadb;
  color: black;
}

#rStudioHeader #menu .menuItem:hover {
  background-color: white;
  color: #4254A7;
}

#rStudioHeader #menu .menuItem.current {
  background-color: #fcfcfc;
  color: #4254A7;
  text-decoration: none;
}

/* Footer */

#rStudioFooter.band {
  background-color: #4254a730;
}

#rStudioFooter .bandContent #copyright {
  color: #4254A7;
}

/* Tables */

table tbody tr:nth-child(even){
  background-color: #fcfcfc;
}

table tbody tr:nth-child(odd){
  background-color: #1a162d10;
}

.latest {
  border-top: .5em solid #dae8f6;
}


/* ------Accent Font------- */



  h1, h2, 
  .article-title:not(a), /* Title on single pages, but not the Learn list article links*/
  div.tagline, /* homepage bands */
  #TableOfContents a, /* Sidebar on regular pages*/
  .articleCategory a, /* Sidebar on Blog & Learn list pages */
  .column75 .sectionTitle:not(.back-to), /* e.g. Blog list title, NOT "back to learn" */
  .column25-left .section:nth-child(n+2) .sectionTitle, /* Tutorial sidebar, all but 1st child */
  .column16-left .section:nth-child(n+2) .sectionTitle,
  a.learn-top-nav-item,  /* Bulleted toc items on Learn list page */
  .itemTitle, /* Blog list page titles & Learn List page subsection Titles */
  .tags-list a    /* taxonomy tags on Tutorial page, e.g. packages */                
                            { 
    font-family: 'Libre Baskerville' ;
  }





/* ----Accent color for headers, toc, etc.----- */



  h1,
  div.sectionTitle {
    color: #4254A7;
  }
  
  /* Sidebar links */
  
  a.learn-top-nav-item, /* Learn list page bulleted toc */
  #TableOfContents a,
  .articleCategory a,
  .column25-left .section:nth-child(n+2) a.sectionTitle,
  .column16-left .section:nth-child(n+2) a.sectionTitle { 
    color: #4254A7;
    opacity: .8;
  }
  
  .column25-left .section:nth-child(n+2) a.sectionTitle.current,
  .column16-left .section:nth-child(n+2) a.sectionTitle.current { 
    color: #4254A7;
  }
  
  a.learn-top-nav-item:hover,
  #TableOfContents a:hover,
  .articleCategory a:hover,
  .column25-left .section:nth-child(n+2) a.sectionTitle:hover,
  .column16-left .section:nth-child(n+2) a.sectionTitle:hover {
    color: #333850eb; 
    opacity: 95%;
    text-decoration: none;
  }
  
  .column25-left .sectionTitle.start:first-of-type a,
  .column16-left .sectionTitle.start:first-of-type a {
    color: #4254A7;
  }
  
  .column25-left .sectionTitle.current,
  .column16-left .sectionTitle.current {
    color: #4254A7;
  }
  
  /* headers and accent colors */
  
  .article-title,             /* Most page titles */
  .column75 div.sectionTitle {  /* Learn page section title */
    color: #4254A7;
  }
  
  h2 {
    color: #333850eb;
  }
  
  .itemTitle a { /* For Learn Page headers */
    color: #333850eb;
  }
  
  .itemTitle a:hover { /* For Learn List Page headers */
    color: #333850eb;
    opacity: 80%;
    text-decoration: none;
  }
  
  h3 {
    color: #333850eb;
    opacity: 95%;
  }
  
  h3 a {
    color: #4254A7;
    /*text-transform: none;
    font-size: 1.5em;
    letter-spacing: normal; */ /* was in tm, but check if i need here? */
  }
  
  h4 {
    color: #333850eb;
    opacity: 90%;
  }
  
  /* taxonomy tags, e.g. tutorial package links*/
  
  .tags-list .learning-objective {
    color: #333850eb;
    opacity: 85%;
  }
  
  .tags-list a {
    color: #333850eb;
    opacity: 56%;
  }
  
  .tags-list a:hover {
    opacity: 95%;
  }



/* Custom colored bullets */



  /*.article-content ul{
    list-style: none;
  } */

  .article-content ul li::before {
  	content: "\2022";
  	color: #f6b73c ;
  	display: inline-block;
  	width: 1.5em;
  	margin-left: -1.7em;
  }
  	
  .article-content ul > li > p {
    display: inline;
  }

  .article-content ul > li { /*to add back in space between bullet items that I took away by making p display inline */
    margin-top: 1em;
    margin-bottom: 1em;
  }
  
  .learn-top-nav ul li::before, /* For learn list page */
  .tutorial #TableOfContents ul li::before { /* For tutorials toc bullets */
  	content: "\2022";
  	color: #f6b73c ;
  	display: inline-block;
  	width: 1.5em;
  	margin-left: -1.8em; 
  }

.article-content ul, /* Regular lists */
.learn-top-nav ul,
.tutorial #TableOfContents ul {
  	list-style: none;
  	line-height: 1.6em;
  	padding-left: 2.5em;
  	/* margin-bottom: 0; */
  }
  
/* But also give div tip outline the same color as bullets*/

div.note, 
div.rstudio-tip, 
div.warning {
  border-color: #f6b73c ;
}
    



/* ------Capitalization of Headers------- */


  
  div.tagline {
    font-size: 1.2rem !important;
    text-transform: uppercase;
    font-weight: bold;
    letter-spacing: 2pt;
  }
  
  #menu {
    text-transform: uppercase;
    letter-spacing: 2pt;
    font-size: 0.9rem !important;
  }
  
  .column25 .sectionTitle,
  .column25 .sectionTitle a,
  .column16 .sectionTitle,
  .column16 .sectionTitle a {
    text-transform: uppercase;
    letter-spacing: 2pt;
    font-size: 0.8rem;
    font-weight: bold;
  }
  
  .icon-attribution, 
  .icon-attribution a {
    text-transform: uppercase;
  }
  
  /* Getting Started / Tutorial sidebar title */
  .column25-left .sectionTitle.start:first-of-type a,
  .column16-left .sectionTitle.start:first-of-type a {
    text-transform: uppercase;
    letter-spacing: 2pt;
    font-weight: bold;
    font-size: 0.8rem;
  }
  
  /*.e.g "Back to Learn" link */
  .column75 .section .sectionTitle a {
    text-transform: uppercase;
    letter-spacing: 2pt;
    font-weight: bold;
    font-size: 0.8rem;
  }
  
  /*  */
  

  

/* -----Hexes------*/


  #homeContent .band.first .bandContent {
    background-image: url(../css/images/tidy-back-01-thin.svg);
  }
  
  #homeContent .band.second .bandContent {
    background-image: url(../css/images/tidy-back-02-thin.svg);
  }
  
  #homeContent .band.third .bandContent {
    background-image: url(../css/images/tidy-back-03-thin.svg);
  }
  
  .package-section .packages .package {
    background-image: url(../css/images/tidy-packages-back-01-thin.svg); 
  }

  .package-section .packages .package:first-of-type {
      background-image: url(../css/images/tidy-packages-back-02-thin.svg); 
  }

  .package-section .packages .package:last-of-type {
      background-image: url(../css/images/tidy-packages-back-03-thin.svg); 
  } 



/*----- Borders------ */



  #rStudioHeader {
    box-shadow: 0 1px 0px 0px rgba(0, 0, 0, 0.15);
    -moz-box-shadow: 0 1px 0px 0px rgba(0, 0, 0, 0.15);
    -webkit-box-shadow: 0 1px 0px 0px rgba(0, 0, 0, 0.15);
  }
  
  #homeContent .band.second {
    border-bottom: solid 1px #d9d9d9;
    border-top: solid 1px #d9d9d9;
  }
  
  .column25,
  .column16 {
    border-left: solid #d9d9d9 1px;
  }
  
  .column25-left {
    padding-right: 3em;
    border-right: solid #d9d9d9 1px;
}

  .column16-left {
    padding-right: 1.5em;
    border-right: solid #d9d9d9 1px;
}

  .article-content.learn-list ul {
      border-left: 1px solid #d9d9d9;
  }



/*--- Event Stickies ---*/


  .event {
    -moz-box-shadow: 0 0px 0px 0px rgba(0, 0, 0, 0.1);
    -webkit-box-shadow: 0 0px 0px 0px rgba(0, 0, 0, 0.1);
    box-shadow: 0 0px 0px 0px rgba(0, 0, 0, 0.1);
  }




  .event:nth-child(2) {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg); 
  }
  .event:nth-child(3) {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg); 
  }




  .section .event {
    background-color: #eab0c41a;
  }




  .section .event {
    border-top: 7pt solid #a19ea936;
  }




  .section .event a{
      color: #1a162d;
  }




  .section .event {
      color: #404040;
  }


/* ------Test Drive Button-------*/














 

</style>

  <link rel="stylesheet" href="/css/">




<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="generator" content="Hugo 0.68.3" />
<title>Nutriverse</title>
<meta property="og:title" content="Statistical analysis of contingency tables" />
<meta property="og:description" content="Use tests of independence and goodness of fit to analyze tables of counts.
" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://nutriverse.io/learn/statistics/xtabs/" />



<meta property="twitter:card" content="summary_large_image">



<meta property="og:title" content="Statistical analysis of contingency tables" />
<meta property="og:description" content="Use tests of independence and goodness of fit to analyze tables of counts.
" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://nutriverse.io/learn/statistics/xtabs/" />


  <meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Statistical analysis of contingency tables"/>
<meta name="twitter:description" content="Use tests of independence and goodness of fit to analyze tables of counts.
"/>



<link rel="apple-touch-icon" sizes="180x180" href="/images/favicons/apple-touch-icon.png" />
<link rel="icon" type="image/png" sizes="32x32" href="/images/favicons/favicon-32x32.png" />
<link rel="icon" type="image/png" sizes="16x16" href="/images/favicons/favicon-16x16.png" />
<link rel="manifest" href="/images/favicons/site.webmanifest" />
<link rel="mask-icon" href="/images/favicons/safari-pinned-tab.svg" />
<link rel="shortcut icon" href="/images/favicons/favicon.ico" />
<meta name="msapplication-TileColor" content="#da532c">
<meta name="msapplication-config" href="/images/favicons/browserconfig.xml" >



<script type="text/javascript" src="/js/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="/js/site.js"></script>



<link rel="icon" href="/images/favicon.ico" />


  
<script type="text/javascript" src="/js/clipboard.js"></script>






  


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>


<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

  







  <body>
    <div id="appTidyverseSite" class="shrinkHeader alwaysShrinkHeader">
      <div id="main">
        
        <div id="rStudioHeader">
          <div class="band">
            <div class="innards bandContent">
              <div>
                
                    <a href="/" class="nav-logo">
                      <img src="/images/leaf.png" width="37" height="40" alt="Nutriverse">
                    </a>
              </div>
              <div id="menu">
  <div id="menuToggler"></div>
  <div id="menuItems" class="">
    
    
      
      
      
    <a class="menuItem " href="/packages/">Packages</a>
    
      
      
      
    <a class="menuItem " href="/blog/">Blog</a>
    
  </div>
</div>

            </div>
          </div>
        </div>
</head>
<body>


 




<div class="band padForHeader pushFooter">
  <div class="bandContent">
    <div class="full splitColumns withMobileMargins">
      <div class="column75">

      <h1 class="article-title">Statistical analysis of contingency tables</h1>

      
      <div class="article-header" aria-hidden="true">
          
        
      </div>

      
      
      
      
        </p>

      <div class="article-content">
      <h2 id="introduction">Introduction</h2>
<p>This article only requires that you have the tidymodels package installed.</p>
<p>In this vignette, we&rsquo;ll walk through conducting a <code>\(\chi^2\)</code> (chi-squared) test of independence and a chi-squared goodness of fit test using infer. We&rsquo;ll start out with a chi-squared test of independence, which can be used to test the association between two categorical variables. Then, we&rsquo;ll move on to a chi-squared goodness of fit test, which tests how well the distribution of one categorical variable can be approximated by some theoretical distribution.</p>
<p>Throughout this vignette, we&rsquo;ll make use of the <code>ad_data</code> data set (available in the modeldata package, which is part of tidymodels). This data set is related to cognitive impairment in 333 patients from 
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3079734/" target="_blank" rel="noopener">Craig-Schapiro <em>et al</em> (2011)</a>. See <code>?ad_data</code> for more information on the variables included and their source. One of the main research questions in these data were how a person&rsquo;s genetics related to the Apolipoprotein E gene affect their cognitive skills. The data shows:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#00f">library</span>(tidymodels) <span style="color:#408080;font-style:italic"># Includes the infer package</span>

<span style="color:#00f">data</span>(ad_data, package <span style="color:#666">=</span> <span style="color:#ba2121">&#34;modeldata&#34;</span>)
ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">select</span>(Genotype, Class)
<span style="color:#408080;font-style:italic">#&gt; # A tibble: 333 x 2</span>
<span style="color:#408080;font-style:italic">#&gt;    Genotype Class   </span>
<span style="color:#408080;font-style:italic">#&gt;    &lt;fct&gt;    &lt;fct&gt;   </span>
<span style="color:#408080;font-style:italic">#&gt;  1 E3E3     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  2 E3E4     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  3 E3E4     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  4 E3E4     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  5 E3E3     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  6 E4E4     Impaired</span>
<span style="color:#408080;font-style:italic">#&gt;  7 E2E3     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  8 E2E3     Control </span>
<span style="color:#408080;font-style:italic">#&gt;  9 E3E3     Control </span>
<span style="color:#408080;font-style:italic">#&gt; 10 E2E3     Impaired</span>
<span style="color:#408080;font-style:italic">#&gt; # … with 323 more rows</span>
</code></pre></div><p>The three main genetic variants are called E2, E3, and E4. The values in <code>Genotype</code> represent the genetic makeup of patients based on what they inherited from their parents (i.e, a value of &ldquo;E2E4&rdquo; means E2 from one parent and E4 from the other).</p>
<h2 id="test-of-independence">Test of independence</h2>
<p>To carry out a chi-squared test of independence, we&rsquo;ll examine the association between their cognitive ability (impaired and healthy) and the genetic makeup. This is what the relationship looks like in the sample data:</p>
<p><img src="figs/plot-indep-1.svg" width="672" /></p>
<p>If there were no relationship, we would expect to see the purple bars reaching to the same length, regardless of cognitive ability. Are the differences we see here, though, just due to random noise?</p>
<p>First, to calculate the observed statistic, we can use <code>specify()</code> and <code>calculate()</code>.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># calculate the observed statistic</span>
observed_indep_statistic <span style="color:#666">&lt;-</span> ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(Genotype <span style="color:#666">~</span> Class) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">calculate</span>(stat <span style="color:#666">=</span> <span style="color:#ba2121">&#34;Chisq&#34;</span>)
</code></pre></div><p>The observed <code>\(\chi^2\)</code> statistic is 21.577. Now, we want to compare this statistic to a null distribution, generated under the assumption that these variables are not actually related, to get a sense of how likely it would be for us to see this observed statistic if there were actually no association between cognitive ability and genetics.</p>
<p>We can <code>generate()</code> the null distribution in one of two ways: using randomization or theory-based methods. The randomization approach permutes the response and explanatory variables, so that each person&rsquo;s genetics is matched up with a random cognitive rating from the sample in order to break up any association between the two.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># generate the null distribution using randomization</span>
null_distribution_simulated <span style="color:#666">&lt;-</span> ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(Genotype <span style="color:#666">~</span> Class) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">hypothesize</span>(null <span style="color:#666">=</span> <span style="color:#ba2121">&#34;independence&#34;</span>) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">generate</span>(reps <span style="color:#666">=</span> <span style="color:#666">5000</span>, type <span style="color:#666">=</span> <span style="color:#ba2121">&#34;permute&#34;</span>) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">calculate</span>(stat <span style="color:#666">=</span> <span style="color:#ba2121">&#34;Chisq&#34;</span>)
</code></pre></div><p>Note that, in the line <code>specify(Genotype ~ Class)</code> above, we could use the equivalent syntax <code>specify(response = Genotype, explanatory = Class)</code>. The same goes in the code below, which generates the null distribution using theory-based methods instead of randomization.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># generate the null distribution by theoretical approximation</span>
null_distribution_theoretical <span style="color:#666">&lt;-</span> ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(Genotype <span style="color:#666">~</span> Class) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">hypothesize</span>(null <span style="color:#666">=</span> <span style="color:#ba2121">&#34;independence&#34;</span>) <span style="color:#666">%&gt;%</span>
  <span style="color:#408080;font-style:italic"># note that we skip the generation step here!</span>
  <span style="color:#00f">calculate</span>(stat <span style="color:#666">=</span> <span style="color:#ba2121">&#34;Chisq&#34;</span>)
</code></pre></div><p>To get a sense for what these distributions look like, and where our observed statistic falls, we can use <code>visualize()</code>:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># visualize the null distribution and test statistic!</span>
null_distribution_simulated <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">visualize</span>() <span style="color:#666">+</span> 
  <span style="color:#00f">shade_p_value</span>(observed_indep_statistic,
                direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)
</code></pre></div><p><img src="figs/visualize-indep-1.svg" width="672" /></p>
<p>We could also visualize the observed statistic against the theoretical null distribution. Note that we skip the <code>generate()</code> and <code>calculate()</code> steps when using the theoretical approach, and that we now need to provide <code>method = &quot;theoretical&quot;</code> to <code>visualize()</code>.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># visualize the theoretical null distribution and test statistic!</span>
ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(Genotype <span style="color:#666">~</span> Class) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">hypothesize</span>(null <span style="color:#666">=</span> <span style="color:#ba2121">&#34;independence&#34;</span>) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">visualize</span>(method <span style="color:#666">=</span> <span style="color:#ba2121">&#34;theoretical&#34;</span>) <span style="color:#666">+</span> 
  <span style="color:#00f">shade_p_value</span>(observed_indep_statistic,
                direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)
</code></pre></div><p><img src="figs/visualize-indep-theor-1.svg" width="672" /></p>
<p>To visualize both the randomization-based and theoretical null distributions to get a sense of how the two relate, we can pipe the randomization-based null distribution into <code>visualize()</code>, and further provide <code>method = &quot;both&quot;</code>.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># visualize both null distributions and the test statistic!</span>
null_distribution_simulated <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">visualize</span>(method <span style="color:#666">=</span> <span style="color:#ba2121">&#34;both&#34;</span>) <span style="color:#666">+</span> 
  <span style="color:#00f">shade_p_value</span>(observed_indep_statistic,
                direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)
</code></pre></div><p><img src="figs/visualize-indep-both-1.svg" width="672" /></p>
<p>Either way, it looks like our observed test statistic would be fairly unlikely if there were actually no association between cognition and genotype. More exactly, we can calculate the p-value:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># calculate the p value from the observed statistic and null distribution</span>
p_value_independence <span style="color:#666">&lt;-</span> null_distribution_simulated <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">get_p_value</span>(obs_stat <span style="color:#666">=</span> observed_indep_statistic,
              direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)

p_value_independence
<span style="color:#408080;font-style:italic">#&gt; # A tibble: 1 x 1</span>
<span style="color:#408080;font-style:italic">#&gt;    p_value</span>
<span style="color:#408080;font-style:italic">#&gt;      &lt;dbl&gt;</span>
<span style="color:#408080;font-style:italic">#&gt; 1 0.000600</span>
</code></pre></div><p>Thus, if there were really no relationship between cognition and genotype, the probability that we would see a statistic as or more extreme than 21.577 is approximately 6\times 10^{-4}.</p>
<p>Note that, equivalently to the steps shown above, the package supplies a wrapper function, <code>chisq_test</code>, to carry out Chi-Squared tests of independence on tidy data. The syntax goes like this:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#00f">chisq_test</span>(ad_data, Genotype <span style="color:#666">~</span> Class)
<span style="color:#408080;font-style:italic">#&gt; # A tibble: 1 x 3</span>
<span style="color:#408080;font-style:italic">#&gt;   statistic chisq_df  p_value</span>
<span style="color:#408080;font-style:italic">#&gt;       &lt;dbl&gt;    &lt;int&gt;    &lt;dbl&gt;</span>
<span style="color:#408080;font-style:italic">#&gt; 1      21.6        5 0.000630</span>
</code></pre></div><h2 id="goodness-of-fit">Goodness of fit</h2>
<p>Now, moving on to a chi-squared goodness of fit test, we&rsquo;ll take a look at just the genotype data. Many papers have investigated the relationship of Apolipoprotein E to diseases. For example, 
<a href="https://annals.org/aim/article-abstract/717641/meta-analysis-apolipoprotein-e-genotypes-risk-coronary-heart-disease" target="_blank" rel="noopener">Song <em>et al</em> (2004)</a> conducted a meta-analysis of numerous studies that looked at this gene and heart disease. In their paper, they describe the frequency of the different genotypes across many samples. For the cognition study, it might be interesting to see if our sample of genotypes was consistent with this literature (treating the rates, for this analysis, as known).</p>
<p>The rates of the meta-analysis and our observed data are:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># Song, Y., Stampfer, M. J., &amp; Liu, S. (2004). Meta-Analysis: Apolipoprotein E </span>
<span style="color:#408080;font-style:italic"># Genotypes and Risk for Coronary Heart Disease. Annals of Internal Medicine, </span>
<span style="color:#408080;font-style:italic"># 141(2), 137.</span>
meta_rates <span style="color:#666">&lt;-</span> <span style="color:#00f">c</span>(<span style="color:#ba2121">&#34;E2E2&#34;</span> <span style="color:#666">=</span> <span style="color:#666">0.71</span>, <span style="color:#ba2121">&#34;E2E3&#34;</span> <span style="color:#666">=</span> <span style="color:#666">11.4</span>, <span style="color:#ba2121">&#34;E2E4&#34;</span> <span style="color:#666">=</span> <span style="color:#666">2.32</span>,
                <span style="color:#ba2121">&#34;E3E3&#34;</span> <span style="color:#666">=</span> <span style="color:#666">61.0</span>, <span style="color:#ba2121">&#34;E3E4&#34;</span> <span style="color:#666">=</span> <span style="color:#666">22.6</span>, <span style="color:#ba2121">&#34;E4E4&#34;</span> <span style="color:#666">=</span> <span style="color:#666">2.22</span>)
meta_rates <span style="color:#666">&lt;-</span> meta_rates<span style="color:#666">/</span><span style="color:#00f">sum</span>(meta_rates) <span style="color:#408080;font-style:italic"># these add up to slightly &gt; 100%</span>

obs_rates <span style="color:#666">&lt;-</span> <span style="color:#00f">table</span>(ad_data<span style="color:#666">$</span>Genotype)<span style="color:#666">/</span><span style="color:#00f">nrow</span>(ad_data)
<span style="color:#00f">round</span>(<span style="color:#00f">cbind</span>(obs_rates, meta_rates) <span style="color:#666">*</span> <span style="color:#666">100</span>, <span style="color:#666">2</span>)
<span style="color:#408080;font-style:italic">#&gt;      obs_rates meta_rates</span>
<span style="color:#408080;font-style:italic">#&gt; E2E2       0.6       0.71</span>
<span style="color:#408080;font-style:italic">#&gt; E2E3      11.1      11.37</span>
<span style="color:#408080;font-style:italic">#&gt; E2E4       2.4       2.31</span>
<span style="color:#408080;font-style:italic">#&gt; E3E3      50.1      60.85</span>
<span style="color:#408080;font-style:italic">#&gt; E3E4      31.8      22.54</span>
<span style="color:#408080;font-style:italic">#&gt; E4E4       3.9       2.21</span>
</code></pre></div><p>Suppose our null hypothesis is that <code>Genotype</code> follows the same frequency distribution as the meta-analysis. Lets now test whether this difference in distributions is statistically significant.</p>
<p>First, to carry out this hypothesis test, we would calculate our observed statistic.</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># calculating the null distribution</span>
observed_gof_statistic <span style="color:#666">&lt;-</span> ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(response <span style="color:#666">=</span> Genotype) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">hypothesize</span>(null <span style="color:#666">=</span> <span style="color:#ba2121">&#34;point&#34;</span>, p <span style="color:#666">=</span> meta_rates) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">calculate</span>(stat <span style="color:#666">=</span> <span style="color:#ba2121">&#34;Chisq&#34;</span>)
</code></pre></div><p>The observed statistic is 23.384. Now, generating a null distribution, by just dropping in a call to <code>generate()</code>:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># generating a null distribution</span>
null_distribution_gof <span style="color:#666">&lt;-</span> ad_data <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">specify</span>(response <span style="color:#666">=</span> Genotype) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">hypothesize</span>(null <span style="color:#666">=</span> <span style="color:#ba2121">&#34;point&#34;</span>, p <span style="color:#666">=</span> meta_rates) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">generate</span>(reps <span style="color:#666">=</span> <span style="color:#666">5000</span>, type <span style="color:#666">=</span> <span style="color:#ba2121">&#34;simulate&#34;</span>) <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">calculate</span>(stat <span style="color:#666">=</span> <span style="color:#ba2121">&#34;Chisq&#34;</span>)
</code></pre></div><p>Again, to get a sense for what these distributions look like, and where our observed statistic falls, we can use <code>visualize()</code>:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># visualize the null distribution and test statistic!</span>
null_distribution_gof <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">visualize</span>() <span style="color:#666">+</span> 
  <span style="color:#00f">shade_p_value</span>(observed_gof_statistic,
                direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)
</code></pre></div><p><img src="figs/visualize-indep-gof-1.svg" width="672" /></p>
<p>This statistic seems like it would be unlikely if our rates were the same as the rates from the meta-analysis! How unlikely, though? Calculating the p-value:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#408080;font-style:italic"># calculate the p-value</span>
p_value_gof <span style="color:#666">&lt;-</span> null_distribution_gof <span style="color:#666">%&gt;%</span>
  <span style="color:#00f">get_p_value</span>(observed_gof_statistic,
              direction <span style="color:#666">=</span> <span style="color:#ba2121">&#34;greater&#34;</span>)

p_value_gof
<span style="color:#408080;font-style:italic">#&gt; # A tibble: 1 x 1</span>
<span style="color:#408080;font-style:italic">#&gt;   p_value</span>
<span style="color:#408080;font-style:italic">#&gt;     &lt;dbl&gt;</span>
<span style="color:#408080;font-style:italic">#&gt; 1  0.0008</span>
</code></pre></div><p>Thus, if each genotype occurred at the same rate as the Song paper, the probability that we would see a distribution like the one we did is approximately 8\times 10^{-4}.</p>
<p>Again, equivalently to the steps shown above, the package supplies a wrapper function, <code>chisq_test</code>, to carry out chi-squared goodness of fit tests on tidy data. The syntax goes like this:</p>
<div class="highlight"><pre style=";-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r"><span style="color:#00f">chisq_test</span>(ad_data, response <span style="color:#666">=</span> Genotype, p <span style="color:#666">=</span> meta_rates)
<span style="color:#408080;font-style:italic">#&gt; # A tibble: 1 x 3</span>
<span style="color:#408080;font-style:italic">#&gt;   statistic chisq_df  p_value</span>
<span style="color:#408080;font-style:italic">#&gt;       &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;</span>
<span style="color:#408080;font-style:italic">#&gt; 1      23.4        5 0.000285</span>
</code></pre></div><h2 id="session-information">Session information</h2>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 3.6.2 (2019-12-12)
#&gt;  os       macOS Mojave 10.14.6        
#&gt;  system   x86_64, darwin15.6.0        
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       America/Denver              
#&gt;  date     2020-04-17                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────
#&gt;  package    * version date       lib source        
#&gt;  broom      * 0.5.5   2020-02-29 [1] CRAN (R 3.6.0)
#&gt;  dials      * 0.0.6   2020-04-03 [1] CRAN (R 3.6.2)
#&gt;  dplyr      * 0.8.5   2020-03-07 [1] CRAN (R 3.6.0)
#&gt;  ggplot2    * 3.3.0   2020-03-05 [1] CRAN (R 3.6.0)
#&gt;  infer      * 0.5.1   2019-11-19 [1] CRAN (R 3.6.0)
#&gt;  parsnip    * 0.1.0   2020-04-09 [1] CRAN (R 3.6.2)
#&gt;  purrr      * 0.3.3   2019-10-18 [1] CRAN (R 3.6.0)
#&gt;  recipes    * 0.1.10  2020-03-18 [1] CRAN (R 3.6.0)
#&gt;  rlang        0.4.5   2020-03-01 [1] CRAN (R 3.6.0)
#&gt;  rsample    * 0.0.6   2020-03-31 [1] CRAN (R 3.6.2)
#&gt;  tibble     * 2.1.3   2019-06-06 [1] CRAN (R 3.6.2)
#&gt;  tidymodels * 0.1.0   2020-02-16 [1] CRAN (R 3.6.0)
#&gt;  tune       * 0.1.0   2020-04-02 [1] CRAN (R 3.6.2)
#&gt;  workflows  * 0.1.1   2020-03-17 [1] CRAN (R 3.6.0)
#&gt;  yardstick  * 0.0.6   2020-03-17 [1] CRAN (R 3.6.0)
#&gt; 
#&gt; [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
</code></pre>
      </div>


      


      </div> 
      
      <div class="column25">
        <div class="section hideOnMobile">
          
          <div class="sectionTitle">Contents</div>
            <nav id="TableOfContents">
  <ul>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#test-of-independence">Test of independence</a></li>
    <li><a href="#goodness-of-fit">Goodness of fit</a></li>
    <li><a href="#session-information">Session information</a></li>
  </ul>
</nav>
          
        </div>
        <div class="section">
          





        </div>
      </div> 

    </div> 
  </div>  
</div> 



        <div id="rStudioFooter" class="band">
          <div class="bandContent">
            <div id="copyright">
              
              © <a href="https://katilingban.io/">Katilingban</a>, 2020<br>
              
              
            </div>
            <div id="logos">
              <a href="https://github.com/nutriverse" class="footerLogo gitHub" aria-label="Tidyverse GitHub organization"></a>
              <a href="https://twitter.com/katilingban" class="footerLogo twitter" aria-label="rstats hashtag on twitter.com"></a>
            </div>
          </div>
        </div>

      </div>  
    </div>  

    

    
<script src="/js/math-code.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>


    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'G-Z9SDQC2LZH', 'auto');
	
	ga('send', 'pageview');
}
</script>


  </body>
</html>

</body>
</html>
